src/scene.rs

```diff
@@ -79,14 +79,17 @@ impl Transform {
                     mem::replace(&mut m[3][2], 0.0),
                 ];
                 let sx = Vector3::new(m[0][0], m[0][1], m[0][2]).magnitude();
+                let sy = Vector3::new(m[1][0], m[1][1], m[1][2]).magnitude();
+                let mut sz = Vector3::new(m[2][0], m[2][1], m[2][2]).magnitude();
+                if cgmath::Matrix4::from(m).determinant() < 0.0 {
+                    sz = -sz;
```

Comment:
Is any further negation required here?

Reply:
No, that looks good. If you want, you can cut `m` down to a `Matrix3` right after you extract the translation though, instead of when you form `r`, which might make taking the determinant a little faster and the code a little nicer (eg. `let sx = m.x.magnitude()`, `m.x /= sx`).

Reply:
Nice, will do.

Reply:
Using `signum` is very pretty!